captum/attr/_utils/lrp_rules.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+
+from abc import ABC, abstractmethod
+
+import torch
+
+
+class PropagationRule(ABC):
+    """
+    Base class for all propagation rule classes, also called Z-Rule.
+    STABILITY_FACTOR is used to assure that no zero divison occurs.
+    """
+
+    STABILITY_FACTOR = 1e-9
+
+    def forward_hook(self, module, inputs, outputs):
+        """Register backward hooks on input and output
+        tensors of linear layers in the model."""
+        self._handle_input_hooks = []
+        for input in inputs:
```

Comment:
A question: Do we assume that inputs are always a tuple here ?

Reply:
Yes, I think that is the assumption.

Reply:
Yes, I think that is the assumption here.

Reply:
I have added `_format_tensor_into_tuples()` now

Reply:
I added the function call now.